scripts/migrate_locales.js

```diff
@@ -0,0 +1,366 @@
+      case '-v':
+        Object.assign(PARSED_ARGS, { verbose: true })
+        break
+
+      case '--version':
+      case '-V':
+        printVersion()
+        break
+
+      case '--help':
+      case '-h':
+        printHelp()
+        break
+    }
+  }
+
+  validateOptions(PARSED_ARGS)
+}
+
+function deleteAtKey({ object, keyPath, level = 0 }) {
+  if (!object) {
+    return {}
+  }
+
+  // Key path is the key itself
+  if (object[keyPath]) {
+    const value = object[keyPath]
+    delete object[keyPath]
+    return { value, pathAsKey: true }
+  }
+
+  // Nested keys
+  const nestedKeys = keyPath.split('.')
+  const key = nestedKeys[level]
+
+  // Last key in keys path
+  if (level >= nestedKeys.length - 1) {
+    const value = object[key]
+
+    if (value) {
+      delete object[key]
+    }
+
+    return { value, pathAsKey: false }
+  }
+
+  const { value, pathAsKey } = deleteAtKey({
+    object: object[key],
+    keyPath,
+    level: level + 1,
+  })
+
+  if (object[key] && !Object.keys(object[key]).length) {
+    delete object[key]
+  }
+
+  return { value, pathAsKey }
+}
+
+function insertAtKey({ object, value, keyPath, level = 0, pathAsKey = false }) {
+  if (!object) {
+    return
+  }
+
+  // Key path is the key itself
+  if (pathAsKey) {
+    object[keyPath] = value
+    return
+  }
+
+  // Nested keys
+  const nestedKeys = keyPath.split('.')
+  const key = nestedKeys[level]
+
+  // Last key in keys path
+  if (level >= nestedKeys.length - 1) {
+    object[key] = value
+    return
+  }
+
+  if (!object[key]) {
+    object[key] = {}
+  }
+
+  insertAtKey({ object: object[key], value, keyPath, level: level + 1 })
+}
+
+/* Main functions */
+function listVersions() {
+  const versions = Object.keys(VERSIONS)
+    .sort()
+    .map((pair) => {
+      const [from, to] = pair.split('_')
+      return `* from ${from} to ${to}`
+    })
+
+  console.info(`\nSupported versions:${['', ...versions].join('\n  ')}`)
+  process.exit(0)
+}
+
+function migrate(object, dataKey) {
+  if (!object || !Object.keys(object).length) {
+    return
+  }
+
+  const { fromVersion, toVersion } = PARSED_ARGS
+  const changeLog = VERSIONS[[fromVersion, toVersion].join('_')]
+
+  verboseLogging(
+    `\nMigrate locale keys for ${buildColorMessage(dataKey, COLORS.BLUE)}:\n`
+  )
+
+  Object.keys(changeLog).forEach((fromKey) => {
+    const { value: possibleValue, pathAsKey } = deleteAtKey({
+      object,
+      keyPath: fromKey,
+    })
+
+    if (!possibleValue) {
+      return
+    }
+
+    const toKeys = changeLog[fromKey]
+
+    verboseLogging(
+      `  - Found obsolete key ${buildColorMessage(
+        fromKey,
+        COLORS.YELLOW
+      )}, replace with:`
+    )
+    toKeys.forEach((toKey) =>
+      verboseLogging('\t*', buildColorMessage(toKey, COLORS.BLUE))
+    )
+
+    toKeys.forEach((toKey) =>
+      insertAtKey({
+        object,
+        value: possibleValue,
+        keyPath: toKey,
+        pathAsKey,
+      })
+    )
+  })
```

Comment:
Why do you need two iterations here? You can just
```suggestion
    toKeys.forEach((toKey) => {
      verboseLogging('\t*', buildColorMessage(toKey, COLORS.BLUE))
      insertAtKey({
        object,
        value: possibleValue,
        keyPath: toKey,
        pathAsKey,
      })
    })
  })
```

Reply:
You're right, good catch!